### PR TITLE
Remove not used QueryChannel in QueryCoord & QueryNode

### LINKED ISSUE #16855: [Enhancement]: Remove QueryChannel after search/query using grpc
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What would you like to be added?

Remove not used `QueryChannel` related logic since search/query request are delivered via Grpc now.

### Why is this needed?

Since query channel are fully replaced by Grpc and segments online/offline control mechanism, removing the querychannel can reduce the topics in mq and operation related.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by congqixia**
Remove QueryChannel related logic:
- WatchQueryChannel call in QueryCoord & QueryNode
- Consume from QueryChannel in QueryNode

Resolves #16855

Signed-off-by: Congqi Xia <congqi.xia@zilliz.com>

### CODE REVIEW

**Comment by congqixia** [comment]
QueryCoord still create QueryChannel producer
/hold

File: internal/querycoord/meta.go [hunk 1]
```diff
@@ -632,37 +630,37 @@ func (m *MetaReplica) saveGlobalSealedSegInfos(saves col2SegmentInfos) (col2Seal
...
-		queryChannelInfo.GlobalSealedSegments = globalSealedSegmentInfos
-		queryChannelInfosMap[collectionID] = queryChannelInfo
-	}
+			globalSealedSegmentInfos := make([]*querypb.SegmentInfo, 0)
+			for _, info := range seg2Info {
+				globalSealedSegmentInfos = append(globalSealedSegmentInfos, info)
+			}
+			queryChannelInfo.GlobalSealedSegments = globalSealedSegmentInfos
+			queryChannelInfosMap[collectionID] = queryChannelInfo
+		}*/
```
Comment (yah01) [inline internal/querycoord/meta.go:663]: these lines (633-663) should be remove?

File: internal/querycoord/meta.go [hunk 2]
```diff
@@ -700,11 +698,12 @@ func (m *MetaReplica) saveGlobalSealedSegInfos(saves col2SegmentInfos) (col2Seal
...
-	for collectionID, channelInfo := range queryChannelInfosMap {
-		m.queryChannelInfos[collectionID] = channelInfo
-	}
-	m.channelMu.Unlock()
+	/*
+		m.channelMu.Lock()
+		for collectionID, channelInfo := range queryChannelInfosMap {
+			m.queryChannelInfos[collectionID] = channelInfo
+		}
+		m.channelMu.Unlock()*/
```
Comment (yah01) [inline internal/querycoord/meta.go:706]: 703-707

File: internal/querycoord/meta.go [hunk 3]
```diff
@@ -733,11 +732,12 @@ func (m *MetaReplica) removeGlobalSealedSegInfos(collectionID UniqueID, partitio
...
-	if err != nil {
-		return nil, err
-	}
-	queryChannelInfo.SeekPosition = msgPosition
+	/*
+		msgPosition, err := m.sendSealedSegmentChangeInfos(collectionID, queryChannelInfo.QueryChannel, segmentChangeInfos)
+		if err != nil {
+			return nil, err
+		}
+		queryChannelInfo.SeekPosition = msgPosition*/
```
Comment (yah01) [inline internal/querycoord/meta.go:740]: 736-740

File: internal/querycoord/meta.go [hunk 4]
```diff
@@ -835,7 +836,7 @@ func (m *MetaReplica) sendSealedSegmentChangeInfos(collectionID UniqueID, queryC
 		ChannelName: queryChannel,
 		MsgID:       messageIDs[0].Serialize(),
 	}, nil
-}
+}*/
```
Comment (yah01) [inline internal/querycoord/meta.go:839]: 791-839

File: internal/querycoord/meta.go [hunk 5]
```diff
@@ -1044,7 +1046,7 @@ func (m *MetaReplica) getQueryStreamByID(collectionID UniqueID, queryChannel str
 	}
 
 	return queryStream, nil
-}
+}*/
```
Comment (yah01) [inline internal/querycoord/meta.go:1049]: 1027-1049

**Review by yah01 — CHANGES_REQUESTED** [review]

File: internal/querycoord/meta.go (hunk 1 above)
Comment (congqixia) [reply internal/querycoord/meta.go:663]: Keep it for a short period of time in case of any meta lost due to removing querychannel.

**Review by yah01 — COMMENTED** [review]
queryStreams & streamMu of MetaReplica in QueryCoord should be removed

**Review by XuanYang-cn — CHANGES_REQUESTED** [review]
Please remove the corresponding `queryNode.msgStream` configs in `milvus.yaml` and `embedding-milvus.yaml`

**Pull request merged**